genesyscloud/telephony_provider_edges_trunkbasesettings/genesyscloud_telephony_providers_trunkbasesettings_schema.go

```diff
@@ -0,0 +1,114 @@
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the entity.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"state": {
+				Description: "The resource's state.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"description": {
+				Description: "The resource's description.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"trunk_meta_base_id": {
+				Description: "The meta-base this trunk is based on.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+
+			"properties": {
+				Description:      "trunk base settings properties",
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				DiffSuppressFunc: util.SuppressEquivalentJsonDiffs,
+			},
+			"trunk_type": {
+				Description:  "The type of this trunk base.Valid values: EXTERNAL, PHONE, EDGE.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"EXTERNAL", "PHONE", "EDGE"}, false),
+			},
+			"managed": {
+				Description: "Is this trunk being managed remotely. This property is synchronized with the managed property of the Edge Group to which it is assigned.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			"inbound_site_id": {
+				Description: "The site to which inbound calls will be routed. Only valid for External BYOC Trunks.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"site_id": {
+				Description: "Used to determine the media regions for inbound and outbound calls through a trunk. Also determines the dial plan to use for calls that came in on a trunk and have to be sent out on it as well.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+		},
+		CustomizeDiff: util.CustomizeTrunkBaseSettingsPropertiesDiff,
+	}
+}
+
+func DataSourceTrunkBaseSettings() *schema.Resource {
+	return &schema.Resource{
+		Description: "Data source for Genesys Cloud Trunk Base Settings. Select a trunk base settings by name",
+		ReadContext: provider.ReadWithPooledClient(dataSourceTrunkBaseSettingsRead),
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Trunk Base Settings name.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+		},
+	}
+}
+
+func TrunkBaseSettingsExporter() *resourceExporter.ResourceExporter {
+	return &resourceExporter.ResourceExporter{
+		GetResourcesFunc: provider.GetAllWithPooledClient(getAllTrunkBaseSettings),
+		RefAttrs:         map[string]*resourceExporter.RefAttrSettings{
+			//"inbound_site_id": {RefType: "genesyscloud_telephony_providers_edges_site"}, TODO: decide how/if this will be included after DEVTOOLING-676 is resolved
```

Comment:
DEVTOOLING-676 is resolved now. Should this be uncommented now?

Reply:
I think we should remove the line instead of uncomment. We closed that ticket because we couldn't get around to it. If we uncomment, then the same unit test will fail and will require something to investigate.

Reply:
I thought we did resolve it? What unit test was failing? If we don't include this, the export won't properly replace with the dynamic reference to the site.

Reply:
I just tested it out and the same unit test is still failing if we uncomment this line. Output:

```bash
=== RUN   TestForExportCycles
    resource_genesyscloud_tf_export_test.go:2363: Found the following potential reference cycles:
         [[genesyscloud_telephony_providers_edges_trunkbasesettings genesyscloud_telephony_providers_edges_site genesyscloud_telephony_providers_edges_trunkbasesettings]]
--- FAIL: TestForExportCycles (0.00s)
FAIL
FAIL    terraform-provider-genesyscloud/genesyscloud/tfexporter 1.040s
FAIL
```